utils/uart_buffer.h

```diff
@@ -0,0 +1,115 @@
+/*
+ * uart_buffer.h
+ *
+ *  Created on: Jan 28, 2017
+ *      Author: Jean-Etienne Tremblay
+ */
+
+#ifndef ZEPHYR_COMMON_API_UART_BUFFER_H_
+#define ZEPHYR_COMMON_API_UART_BUFFER_H_
+
+#include <mbed.h>
+#include "circular_buffer.h"
+
+/** UART Message circular buffer template class + IRQ handler
+ *
+ *  @param TXSize size of receive buffer in messages; must be a power of 2
+ *
+ *  Typical usage:
+ *    // 32 message buffer
+ *    UARTTXBuffer<32> uartBuffer(uart);
+ *
+ *    void handleUartMessage() {
+ *        uartBuffer.handleIrq();
+ *    }
+ *
+ *    void setup() {
+ *       // Do your other setup
+ *       uartBuffer.attach(handleUartMessage);
```

Comment:
This is no longer necessary right? (and all related code that explicitly deals with IRQs)